umodbus/functions.py

```diff
@@ -672,12 +706,18 @@ def create_from_request_pdu(cls, pdu):
 
         fmt = '>' + ('B' * byte_count)
         values = struct.unpack(fmt, pdu[6:])
-        values = [integer_to_binary_list(v) for v in values]
 
-        # Flatten list
-        values = [n for value in values for n in value]
+        res = list()
 
-        return cls(starting_address, quantity, byte_count, values)
+        for i, value in enumerate(values):
+            padding = 8 if (quantity - (8 * i)) // 8 > 0 else quantity % 8
+            fmt = '{{0:0{padding}b}}'.format(padding=padding)
+
+            # Create binary representation of integer, convert it to a list
+            # and reverse the list.
```

Comment:
Why create the binary representatie, why convert it to a list and why reverse it? I can read the code and see it is doing this, but why is this happening?